pennylane/ops/qubit/special_unitary.py

```diff
@@ -0,0 +1,329 @@
+
+    grad_method = None
+    """Gradient computation method."""
+
+    def __init__(self, theta, wires, do_queue=True, id=None):
+        num_wires = 1 if isinstance(wires, int) else len(wires)
+        self.hyperparameters["num_wires"] = num_wires
+        theta_shape = qml.math.shape(theta)
+        expected_dim = 4**num_wires - 1
+
+        if len(theta_shape) not in {1, 2}:
+            raise ValueError(
+                "Expected the parameters to have one or two dimensions without or with "
+                f"broadcasting, respectively. The parameters have shape {theta_shape}"
+            )
+
+        if theta_shape[-1] != expected_dim:
+            raise ValueError(
+                f"Expected the parameters to have shape ({expected_dim},) or (batch_size, "
+                f"{expected_dim}). The parameters have shape {theta_shape}"
+            )
+
+        super().__init__(theta, wires=wires, do_queue=do_queue, id=id)
+
+    @staticmethod
+    def compute_matrix(theta, num_wires):
+        r"""Representation of the operator as a canonical matrix in the computational basis
+        (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.SpecialUnitary.matrix`
+
+        Args:
+            theta (tensor_like): Pauli coordinates of the exponent :math:`A(\theta)`.
+            num_wires (int): The number of wires the matrix acts on.
+
+        Returns:
+            tensor_like: canonical matrix of the special unitary corresponding to ``theta``. It
+                has the shape ``(2**num_wires, 2**num_wires)``.
+
+        Compute the matrix of an element in SU(N), given by the Pauli basis coordinated
+        of the associated Lie algebra element.
+        The :math:`4^n-1` Pauli basis elements of the Lie algebra :math:`\mathfrak{su}(N)`
+        for :math:`n` qubits are
+        :math:`P_m\in\{I, X, Y, Z\}^{\otimes n}\setminus\{I^{\otimes n}\}`, and the special
+        unitary matrix is computed as
+
+        .. math::
+
+            U(\theta) = \exp(i\sum_{m=1}^d \theta_m P_m)
+
+        See :func:`~.ops.qubit.matrix_ops.pauli_basis_strings` for the ordering of Pauli words.
+
+        .. note::
+
+            Note that this method internally handles a complex-valued tensor of size
+            ``(4**num_wires, 2**num_wires, 2**num_wires)``, which requires at least
+            ``4 ** (2 * num_wires - 13)`` GB of memory (at default precision).
+
+        **Example**
+
+        >>> theta = np.array([0.5, 0.1, -0.3])
+        >>> qml.SpecialUnitary.compute_matrix(theta, num_wires=1)
+        array([[ 0.83004499-0.28280371j,  0.0942679 +0.47133952j],
+               [-0.0942679 +0.47133952j,  0.83004499+0.28280371j]])
+        """
+        if num_wires > 5:
```

Comment:
What is the order of time saved here? Also does it depend on the interface?

Reply:
See [here for the specs](https://github.com/PennyLaneAI/pennylane/pull/3650#discussion_r1083793329) of `pauli_basis_matrices` (formerly `pauli_basis`).

This `if` clause mostly avoids cluttering the cache when used with more than five qubits, at the expense of increased computation time.

Reply:
I did not explicitly time the approach in the `if` clause, so I did not look at interfaces yet either. Would there be an alternative?

Reply:
Should something be done about this, still? @rmoyard 
Did I understand the question correctly?

Reply:
Resolving, as there is no alternative currently to this approach if we don't want to clutter the RAM with a large cache entry.

Reply:
All good thanks :+1: